svcdef/svcdef.go

```diff
@@ -0,0 +1,347 @@
+package svcdef
+
+// consider renaming to census, pbregistry, or just to "essence"
+
+import (
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"io"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+type Svcdef struct {
+	// PkgName will be the pacakge name of the go file(s) analyzed. So if a
+	// Go file contained "package authz", then PkgName will be "authz". If
+	// multiple Go files are analyzed, it will be the package name of the last
+	// go file analyzed.
+	PkgName  string
+	Messages []*Message
+	Enums    []*Enum
+	// Service contains the sole service for this Svcdef
+	Service *Service
+}
+
+type Message struct {
+	Name   string
+	Fields []*Field
+}
+
+type Enum struct {
+	Name string
+}
+
+type Map struct {
+	Name      string
+	KeyType   *FieldType
+	ValueType *FieldType
+}
+
+type Service struct {
+	Name    string
+	Methods []*ServiceMethod
+}
+
+type ServiceMethod struct {
+	Name         string
+	RequestType  *FieldType
+	ResponseType *FieldType
+	// Bindings contains information for mapping http paths and paramters onto
+	// the fields of this ServiceMethods RequestType.
+	Bindings []*HTTPBinding
+}
+
+type Field struct {
+	Name string
+	Type *FieldType
+}
+
+type FieldType struct {
+	// Name will contain the name of the type, for example "string" or "bool"
+	Name string
+	// Enum contains a pointer to the Enum type this fieldtype represents, if
+	// this FieldType represents an Enum. If not, Enum is nil.
+	Enum *Enum
+	// Message contains a pointer to the Message type this FieldType
+	// represents, if this FieldType represents a Message. If not, Message is
+	// nil.
+	Message *Message
+	// Map contains a pointer to the Map type this FieldType represents, if
+	// this FieldType represents a Map. If not, Map is nil.
+	Map *Map
+	// StarExpr is True if this FieldType represents a pointer to a type.
+	StarExpr bool
+	// ArrayType is True if this FieldType represents a slice of a type.
+	ArrayType bool
+}
+
+// HTTPBinding represents one of potentially several bindings from a gRPC
+// service method to a particuar HTTP path/verb.
+type HTTPBinding struct {
+	Verb string
+	Path string
+	// There is one HTTPParamter for each of the parent service methods Fields.
```

Comment:
Below you use `Service.RequestType.Fields` and here you say `service method Fields`. The RequestType information is helpful, keep that but make the naming/caps/dots consistent thoughtout documentation.